src/main/java/spoon/reflect/declaration/CtElement.java

```diff
@@ -391,4 +392,18 @@
 	 */
 	List<CtElement> getDirectChildren();
 
+	/**
+	 * @return the source code of this element, with fully-qualified types.
+	 *
```

Comment:
@monperrus, I suggest to add there note like: "Call of toString() method will not modify the spoon model"
Note: It is new and critical contract, which was not true until this PR. But as developer I always expected that toString modifies nothing ... and then I wondered why debugging and looking into model, changes model... :-)